src/macos/gdb.rs

```diff
@@ -459,24 +457,22 @@ impl<'a> Handler for CmdHandler<'a> {
 	/// At most apply one action per thread. GDB likes to send default action for other threads,
 	/// even if it knows only about 1: "vCont;s:1;c" (step thread 1, continue others)
 	fn vcont(&self, actions: Vec<(VCont, Option<ThreadId>)>) -> Result<StopReason, Error> {
-		for (cmd, id) in &actions {
-			let _id = id.unwrap_or(ThreadId {
-				pid: Id::All,
-				tid: Id::All,
-			});
-			//debug!("{:?}", id);
-			//println!(self.tracee.pid());
-			/*match (id.pid, id.tid) {
-				(Id::Id(pid), _) if pid != self.tracee.pid() => continue,
-				(_, Id::Id(tid)) if tid != self.tracee.pid() => continue,
-				(_, _) => (),
-			}*/
-			debug!("vcont: {:?}", *cmd);
-			// need to clone, since std::ops::Range<T: Copy> should probably also be Copy, but it isn't.
-			self.continue_execution(cmd.clone());
-
-			break;
-		}
+		let (cmd, id) = &actions[0];
+
+		let _id = id.unwrap_or(ThreadId {
+			pid: Id::All,
+			tid: Id::All,
+		});
+		//debug!("{:?}", id);
+		//println!(self.tracee.pid());
+		/*match (id.pid, id.tid) {
+			(Id::Id(pid), _) if pid != self.tracee.pid() => continue,
+			(_, Id::Id(tid)) if tid != self.tracee.pid() => continue,
+			(_, _) => (),
+		}*/
+		debug!("vcont: {:?}", *cmd);
+		// need to clone, since std::ops::Range<T: Copy> should probably also be Copy, but it isn't.
+		self.continue_execution(cmd.clone());
```

Comment:
Even though removing the loop is syntactically the same, I guess there was a reason behind this code. @stlankes was the author. Maybe this code is unfinished/buggy.

Reply:
Yeah, I was aware of that. It could be on purpose (see the doc comment) or it could be unfinished code. In either way, having a loop here, that is only executed once, makes the current code less readable in my opinion. Fixing it would be another issue.

https://github.com/hermitcore/uhyve/blob/a12e66548556d988d26d25fd345d7da98c0272f1/src/macos/gdb.rs#L456-L458

Reply:
Could you add a TODO statement here? Or could @stlankes comment on that loop?

Reply:
Doesn't the doc comment clarify the situation? “At most apply one action per thread. GDB likes to send default action for other threads, even if it knows only about 1: "vCont;s:1;c" (step thread 1, continue others)”. So I would assume, this is intended.

Reply:
what about something like this?
```Rust
actions.iter().take(1).for_each(|(cmd, id)| {...})
```
This would then reflect the comment a bit better...

Reply:
All right. Done. :+1: